src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameRequestStream.cs

```diff
@@ -76,8 +76,6 @@ public override int Read(byte[] buffer, int offset, int count)
 #if NET451
         public override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
         {
-            ValidateState(default(CancellationToken));
```

Comment:
It was wrong. This wasn't even looking at the task returned from `Validate` state so was somewhat broken. `ReadAsync` calls `ValidateState`.